doc/reference/random/index.rst

```diff
@@ -0,0 +1,96 @@
+.. _random_reference:
+
+Random
+######
+
+.. toctree::
+   :maxdepth: 2
+
+The random API subsystem provides random number generation APIs in both
+cryptographically and non-cryptographically secure instances. Which
+random API to use is based on the cryptographic requirements of the
+random number. The non-cryptographic APIs will return random values
+much faster if non-cryptographic values are needed.
+
+The cryptographically secure random functions shall be compliant to the
+FIPS 140-2 [NIST02]_ recommended algorithms. Hardware based random-number
+generators (RNG) can be used on platforms with appropriate hardware support.
+Platforms without hardware RNG support shall use the CTR-DRBG algorithm.
+The algorithm can be provided by TinyCrypt or mbedTLS depending on
+your application performance and resource requirements.
+
+  .. note::
+
+    The CTR-DRBG generator needs a cryptographically secure entropy
+    source to maintain cryptographic security of the PRNG.
+
+.. _random_kconfig:
+
+Kconfig Options
+***************
+
+These options can be found in the following path :zephyr_file:`subsys/random/Kconfig`.
+
+:option:`CONFIG_TEST_RANDOM_GENERATOR`
```

Comment:
We should think about how we want to handle testing on platforms that don't have an entropy source. Now that we have non-crypto random generators, it makes sense for those to be available even in a test configuration. I don't think people quite grasp how terrible it is to use a predictable source for randomness when crypto is involved. This should never be used for anything of consequence, and really shouldn't be used then, either.

I have argued before, that if this feature is enabled (or the entropy equivalent), involved crypto code should generate a compile-time failure.

One thing that would help would be to figure out how to get an entropy source to be available within qemu.